pkg/cmd/server/api/types.go

```diff
@@ -279,6 +279,21 @@ type MasterConfig struct {
 
 	// NetworkConfig to be passed to the compiled in network plugin
 	NetworkConfig MasterNetworkConfig
+
+	// JenkinsPipelineConfig holds information about the default Jenkins template
+	// used for JenkinsPipeline build strategy.
+	JenkinsPipelineConfig JenkinsPipelineConfig `json:"jenkinsPipelineConfig"`
```

Comment:
@smarterclayton the consequence of making this optional is:

1) We are not able to determine the default Jenkins service name
2) If you don't provide the namespace/template in master config, we skip instantiation of Jenkins template when you create BC first time and leave Jenkins deployment to user
3) We can't provide the "default" values in master config (each admin will have to add the settings, which might be fine but I am afraid that will decrease the experience a little bit, because in order to use Jenkins Pipeline with automatic Jenkins deployment, you will first have to tweak the master configuration. It might also cause problems with all-in-one setup where you don't have master config, right?